### PR TITLE
Fix error message when Traefik fails to get an IP

### PR DESCRIPTION
**Pull request by kkaempf**
The existing (helpful) message [is never reached](https://github.com/epinio/epinio/issues/187#issuecomment-877022028)

This PR makes the Traefik related error messages more readable and helpful.

See #187 

/cc @jimmykarily since he worked in this area recently.

### CODE REVIEW

File: deployments/traefik.go [hunk 1]
```diff
@@ -128,14 +128,17 @@ func (k Traefik) apply(ctx context.Context, c *kubernetes.Cluster, ui *termui.UI
...
+		return errors.Wrap(err, "failed waiting for Traefik Ingress deployment to exist")
 	}
 	if err := c.WaitForPodBySelectorRunning(ctx, ui, TraefikDeploymentID, "app.kubernetes.io/name=traefik", k.Timeout); err != nil {
-		return errors.Wrap(err, "failed waiting Traefik Ingress deployment to come up")
+		return errors.Wrap(err, "failed waiting for Traefik Ingress deployment to come up")
 	}
 
 	if err := c.WaitUntilServiceHasLoadBalancer(ctx, ui, TraefikDeploymentID, "traefik", duration.ToServiceLoadBalancer()); err != nil {
-		return errors.Wrap(err, "failed waiting for Traefik Ingress to have get a Load Balancer")
+		return errors.Wrap(err, "failed waiting for Traefik Ingress to contact the Load Balancer.\n" +
```
Comment (manno) [inline deployments/traefik.go:138]:
```suggestion
		Timed out waiting for LoadBalancer IP on traefik service
```

**Review by manno — CHANGES_REQUESTED** [review]
This error message is very similar to the one at https://github.com/epinio/epinio/blob/main/internal/cli/clients/install_client.go#L358-L363

I'd prefer the string in a constant in the deployment package, so we remove the duplication.

If you want to use this as an opportunity for a small code cleanup, I'd also switch the code in install_client.go from  `helpers.RunToSuccess` to `c.kubeClient.WaitUntilServiceHasLoadBalancer`.

File: deployments/traefik.go [hunk 2]
```diff
@@ -128,14 +128,17 @@ func (k Traefik) apply(ctx context.Context, c *kubernetes.Cluster, ui *termui.UI
...
-		return errors.Wrap(err, "failed waiting Traefik Ingress deployment to come up")
+		return errors.Wrap(err, "failed waiting for Traefik Ingress deployment to come up")
 	}
 
 	if err := c.WaitUntilServiceHasLoadBalancer(ctx, ui, TraefikDeploymentID, "traefik", duration.ToServiceLoadBalancer()); err != nil {
-		return errors.Wrap(err, "failed waiting for Traefik Ingress to have get a Load Balancer")
+		return errors.Wrap(err, "timed out waiting for LoadBalancer IP on traefik service\n" +
+			"Ensure your kubernetes platform has the ability to provision LoadBalancer IP address.\n\n" +
+			"Follow these steps to enable this ability\n" +
+			"https://github.com/epinio/epinio/blob/main/docs/install.md\n")
```
Comment (jimmykarily) [inline deployments/traefik.go:141]: link is broken after docs restructuring

Comment (kkaempf) [reply deployments/traefik.go:141]:
```suggestion
			"https://github.com/epinio/epinio/blob/main/docs/user/howtos/proivision_externalIP_kube_distros.md\n")
```

Comment (andreas-kupries) [reply deployments/traefik.go]: The `proivision` is a tyop. Should be `provision`.

File: deployments/traefik.go [hunk 3]
```diff
@@ -128,14 +128,17 @@ func (k Traefik) apply(ctx context.Context, c *kubernetes.Cluster, ui *termui.UI
...
-		return errors.Wrap(err, "failed waiting Traefik Ingress deployment to come up")
+		return errors.Wrap(err, "failed waiting for Traefik Ingress deployment to come up")
 	}
 
 	if err := c.WaitUntilServiceHasLoadBalancer(ctx, ui, TraefikDeploymentID, "traefik", duration.ToServiceLoadBalancer()); err != nil {
-		return errors.Wrap(err, "failed waiting for Traefik Ingress to have get a Load Balancer")
+		return errors.Wrap(err, "timed out waiting for LoadBalancer IP on traefik service\n" +
+			"Ensure your kubernetes platform has the ability to provision LoadBalancer IP address.\n\n" +
+			"Follow these steps to enable this ability\n" +
+			"https://github.com/epinio/epinio/blob/main/docs/user/howtos/proivision_externalIP_kube_distros.md\n")
```
Comment (andreas-kupries) [inline deployments/traefik.go:141]:
```suggestion
			"https://github.com/epinio/epinio/blob/main/docs/user/howtos/provision_externalIP_kube_distros.md\n")
```

**Review by andreas-kupries — CHANGES_REQUESTED** [review]

File: deployments/traefik.go (hunk 2 above)
Comment (kkaempf) [reply deployments/traefik.go]: That's how the file is actually named 🤦🏻

Comment (andreas-kupries) [reply deployments/traefik.go]: Oh. Ah well, then a different tyop has been exposed now. Silver lining and all that.

File: deployments/traefik.go (hunk 3 above)
Comment (manno) [inline deployments/traefik.go:141]:
```suggestion
			"https://github.com/epinio/epinio/blob/main/docs/user/howtos/provision_external_ip_for_local_kubernetes.md\n")
```

Comment (kkaempf) [reply deployments/traefik.go]: resolved differently

**Review by manno — APPROVED** [review]

**Comment by manno** [comment]
We should squash and merge after tests are green

**Pull request merged**